Firestore/core/src/model/server_timestamp_util.cc

```diff
@@ -49,12 +51,13 @@ google_firestore_v1_Value EncodeServerTimestamp(
   ++field;
   field->key = nanopb::MakeBytesArray(kLocalWriteTimeKey);
   field->value.which_value_type = google_firestore_v1_Value_timestamp_value_tag;
-  field->value.timestamp_value = local_write_time;
+  field->value.timestamp_value.seconds = local_write_time.seconds();
+  field->value.timestamp_value.nanos = local_write_time.nanoseconds();
 
   if (previous_value) {
     ++field;
     field->key = nanopb::MakeBytesArray(kPreviousValueKey);
-    field->value = *previous_value;
+    field->value = DeepClone(*previous_value);
```

Comment:
It concerns me that we pass these structs around with no wrappers to clarify ownership. Definitely seems like something to address in the future.